pkg/agent/agent.go

```diff
@@ -279,59 +269,28 @@ func (k *klocksmith) process(ctx context.Context) error {
 		klog.Info("Node already marked as unschedulable")
 	}
 
+	drainer := newDrainer(ctx, k.clientset, k.reapTimeout)
+
 	klog.Info("Getting pod list for deletion")
 
-	pods, err := k.getPodsForDeletion(ctx)
-	if err != nil {
-		return fmt.Errorf("getting list of pods for deletion: %w", err)
+	pods, errs := drainer.GetPodsForDeletion(k.nodeName)
+	if len(errs) != 0 {
```

Comment:
While `!=` is correct, `>` is more common and hence idiomatic.
```suggestion
	if len(errs) > 0 {
```